packages/shared/lib/wallet.ts

```diff
@@ -225,8 +226,26 @@ export const api: {
     ),
 } = window['__WALLET_API__']
 
-export const getStoragePath = (appPath: string, profileName: string): string => {
-    return `${appPath}/${WALLET_STORAGE_DIRECTORY}/${profileName}`
+export const getStoragePathParts = async (profileName: string): Promise<{
+    path: string
+    sep: string
+}> => {
+    const userDataPath = await Electron.getUserDataPath()
+    const os = await Electron.getOS()
+    let profileDirName = profileName
+    let sep;
+
+    if (os === 'win32') {
+        sep = '\\'
+        profileDirName = encodeNonASCII(profileDirName)
+    } else {
+        sep = '/'
+    }
+
+    return {
+        path: `${userDataPath}${sep}${WALLET_STORAGE_DIRECTORY}${sep}${profileDirName}`,
```

Comment:
Do we need to do this for the `userDataPath` too? For example if the path is `C:\Users\üsername\AppData\Roaming\Firefly`

Reply:
Oh yeah, actually that means this PR won't work as we don't have control over the replacing user name character, it has to be fixed with the RocksDB version :(

Reply:
Ah that's annoying. I've opened https://github.com/rust-rocksdb/rust-rocksdb/issues/512 to see if there was a reason that flag was removed

Reply:
Although was is also strange is it looks like the wallet is using v0.15 https://github.com/iotaledger/wallet.rs/blob/b68f181604824db86da0a4bef9ad6cb11f3e9949/Cargo.toml#L29
which should have included the fix https://github.com/rust-rocksdb/rust-rocksdb/blob/b91299ca2e2278cd5ed0db35f5687fb5e37118cd/librocksdb-sys/build.rs#L137

Reply:
I wonder if the issue is in wallet.rs or stronghold then 🤔

Reply:
I traced through the code and its definitely in RocksDB its failing, specifically this line https://github.com/facebook/rocksdb/blob/cc1c3ee54eace876ad18c39f931e8e5039823930/port/win/env_win.cc#L1179

Reply:
Managed to build a version 0.16.0 with the `ROCKSDB_WINDOWS_UTF8_FILENAMES` flag set, built wallet.rs to reference, and then used the new version with FF, and it works 🎉 

Although there was some C++ code in rocksdb that wouldn't compile with the flag set, so had to update it 😅

Reply:
Nice! 🎉